CRM/Contribute/Form/Contribution.php

```diff
@@ -650,8 +669,14 @@ public function buildQuickForm() {
       $paymentInstrument = $this->add('select', 'payment_instrument_id',
         ts('Payment Method'),
         array('' => ts('- select -')) + CRM_Contribute_PseudoConstant::paymentInstrument(),
-        TRUE, array('onChange' => "return showHideByValue('payment_instrument_id','{$checkPaymentID}','checkNumber','table-row','select',false);")
+        TRUE
       );
+      $this->addField('card_type', array('entity' => 'FinancialTrxn', 'context' => 'get'));
+      $this->add('text', 'pan_truncation', ts('Card Number'), array(
+        'size' => 5,
+        'maxlength' => 4,
+        'autocomplete' => 'off',
+      ));
     }
```

Comment:
Deja vu - why are we asking to enter the last 4 digits of a credit card in a form - anywhere? In the iATS extension we grab the last 4 digits from the card that's used to pay -> store it in a table -> and then show it on several screens; why do we need an option to enter the last 4 digits manually?